Tests/NIOHTTP1Tests/HTTPServerPipelineHandlerTest.swift

```diff
@@ -706,4 +706,94 @@ class HTTPServerPipelineHandlerTest: XCTestCase {
         XCTAssertFalse(self.channel.isActive)
         self.channel = nil
     }
+
+    func testParserErrorOnly() throws {
+        class VerifyOrderHandler: ChannelInboundHandler {
+            typealias InboundIn = HTTPServerRequestPart
+            typealias OutboundOut = HTTPServerResponsePart
+
+            enum State {
+                case errorExpected
+                case done
+            }
+            var state: State = .errorExpected
+
+            func errorCaught(ctx: ChannelHandlerContext, error: Error) {
+                XCTAssertEqual(HTTPParserError.headerOverflow, error as? HTTPParserError)
+                XCTAssertEqual(.errorExpected, self.state)
+                self.state = .done
+            }
+
+            func channelRead(ctx: ChannelHandlerContext, data: NIOAny) {
+                XCTFail("no requests expected")
+            }
+        }
+
+        let handler = VerifyOrderHandler()
+        XCTAssertNoThrow(try self.channel.pipeline.add(handler: HTTPServerProtocolErrorHandler()).wait())
+        XCTAssertNoThrow(try self.channel.pipeline.add(handler: handler).wait())
+
+        self.channel.pipeline.fireErrorCaught(HTTPParserError.headerOverflow)
+
+        XCTAssertEqual(.done, handler.state)
+    }
+
+    func testLegitRequestFollowedByParserErrorArrivingWhilstReponseOutstanding() throws {
+        func makeRequestHead(uri: String) -> HTTPRequestHead {
+            var requestHead = HTTPRequestHead(version: .init(major: 1, minor: 1), method: .GET, uri: uri)
+            requestHead.headers.add(name: "Host", value: "example.com")
+            return requestHead
+        }
+
+        class VerifyOrderHandler: ChannelInboundHandler {
+            typealias InboundIn = HTTPServerRequestPart
+            typealias OutboundOut = HTTPServerResponsePart
+
+            enum State {
+                case reqHeadExpected
+                case reqEndExpected
+                case errorExpected
+                case done
+            }
+            var state: State = .reqHeadExpected
+
+            func errorCaught(ctx: ChannelHandlerContext, error: Error) {
+                XCTAssertEqual(HTTPParserError.closedConnection, error as? HTTPParserError)
+                XCTAssertEqual(.errorExpected, self.state)
+                self.state = .done
+            }
+
+            func channelRead(ctx: ChannelHandlerContext, data: NIOAny) {
+                switch self.unwrapInboundIn(data) {
+                case .head:
+                    ctx.eventLoop.execute {
```

Comment:
Can you add a comment to explain why this is dispatched on the loop here?

Reply:
addressed